src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Sql/Sprocs/GetNextTask_3.sql

```diff
@@ -0,0 +1,81 @@
+﻿/*************************************************************
+    Stored procedures for get next available task
+**************************************************************/
+--
+-- STORED PROCEDURE
+--     GetNextTask
+--
+-- DESCRIPTION
+--     Get next available task
+--
+-- PARAMETERS
+--     @queueId
+--         * The ID of the task record
+--     @taskHeartbeatTimeoutThresholdInSeconds
+--         * Timeout threshold in seconds for heart keep alive
+--
+CREATE PROCEDURE [dbo].[GetNextTask_3]
+@queueId VARCHAR (64), @taskHeartbeatTimeoutThresholdInSeconds INT=600
+AS
+
+SET NOCOUNT ON;
+DECLARE @lock VARCHAR(200) = 'GetNextTask_Q='+@queueId
+        ,@taskId VARCHAR (64) = NULL
+        ,@expirationDateTime AS DATETIME2 (7)
+        ,@heartbeatDateTime AS DATETIME2 (7) = SYSUTCDATETIME();
+SELECT @expirationDateTime = DATEADD(second, -@taskHeartbeatTimeoutThresholdInSeconds, SYSUTCDATETIME());
+ 
+BEGIN TRY
+    BEGIN TRANSACTION
+
+    EXECUTE sp_getapplock @lock, 'Exclusive'
+
+-- try new tasks first
+    UPDATE T
+      SET Status = 2 -- running
+         ,StartDateTime = SYSUTCDATETIME()
+         ,HeartbeatDateTime = SYSUTCDATETIME()
+         ,Worker = host_name()
+         ,RunId = CAST (NEWID() AS NVARCHAR (50)) 
```

Comment:
>CAST (NEWID() AS NVARCHAR (50))  [](http://example.com/codeflow?start=18&length=32)

Can SQL Server use implicit data type conversion?

Reply:
Do you mean CAST AS? I believe it works same as CONVERT in this case, here is a [reference ](https://docs.microsoft.com/en-us/sql/t-sql/functions/cast-and-convert-transact-sql?redirectedfrom=MSDN&view=sql-server-ver15)

Reply:
You do not need to have explicit cast at all.
For example this
UPDATE TaskInfo SET RunId = newid() WHERE TaskId = 'X'
works.

Reply:
Got it , and it worked in experiment, removed CAST already